core/src/services/swift/pager.rs

```diff
@@ -0,0 +1,179 @@
+    async fn next(&mut self) -> Result<Option<Vec<oio::Entry>>> {
+        if self.done {
+            return Ok(None);
+        }
+
+        let response = self.core.swift_list(&self.path).await?;
+
+        let status_code = response.status();
+
+        if !status_code.is_success() {
+            let error = parse_error(response).await?;
+            return Err(error);
+        }
+
+        let bytes = response.into_body().bytes().await?;
+        let mut decoded_response = serde_json::from_slice::<Vec<ListOpResponse>>(&bytes)
+            .map_err(new_json_deserialize_error)?;
+
+        self.done = true;
+
+        let mut entries = Vec::with_capacity(decoded_response.len());
+
+        while let Some(status) = decoded_response.pop() {
+            let entry: oio::Entry = match status {
+                ListOpResponse::Subdir { subdir } => {
+                    let meta = Metadata::new(EntryMode::DIR);
+                    oio::Entry::new(&subdir, meta)
+                }
+                ListOpResponse::FileInfo {
+                    bytes,
+                    hash,
+                    name,
+                    content_type,
+                    mut last_modified,
+                } => {
+                    // this is the pseudo directory itself; we'll skip it.
+                    if name == self.path {
+                        continue;
+                    }
+
+                    let mut meta = Metadata::new(EntryMode::FILE);
+                    meta.set_content_length(bytes);
+                    meta.set_content_md5(hash.as_str());
+
+                    // we'll change "2023-10-28T19:18:11.682610" to "2023-10-28T19:18:11.682610Z"
+                    last_modified.push('Z');
+                    meta.set_last_modified(parse_datetime_from_rfc3339(last_modified.as_str())?);
+
+                    meta.set_content_type(content_type.as_str());
+
+                    oio::Entry::new(&name, meta)
+                }
+            };
+            entries.push(entry);
+        }
+        Ok(Some(entries))
+    }
+}
+
+#[derive(Debug, Eq, PartialEq, Deserialize)]
+#[serde(untagged)]
+pub(super) enum ListOpResponse {
+    Subdir {
+        subdir: String,
+    },
+    FileInfo {
+        bytes: u64,
+        hash: String,
+        name: String,
+        content_type: String,
+        last_modified: String,
+    },
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn parse_list_response_test() -> Result<()> {
+        let resp = bytes::Bytes::from(
+            r#"
+            [
+                {
+                    "subdir": "animals/"
+                },
+                {
+                    "subdir": "fruit/"
+                },
+                {
+                    "bytes": 147,
+                    "hash": "5e6b5b70b0426b1cc1968003e1afa5ad",
+                    "name": "test.txt",
+                    "content_type": "multipart/form-data;boundary=------------------------25004a866ee9c0cb",
```

Comment:
Where are those examples from? It doesn't make sense for an object to have content-type as `multipart/form-data`.

Reply:
I'm going to take a look into this. I simply created a file and used `list` to get this result.

Reply:
I know what happened. It was my Postman PUT request settings. I've fixed it.